pkg/apis/clusteroperator/types.go

```diff
@@ -48,6 +48,67 @@ type ClusterList struct {
 	Items []Cluster
 }
 
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterVersion represents a version of OpenShift that can be, or is installed.
+type ClusterVersion struct {
+	// +optional
+	metav1.TypeMeta
+	// +optional
+	metav1.ObjectMeta
+
+	// Possible future additions: openshift_release / image_tag / pkg_version.
+
+	YumRepositories []YumRepositories
```

Comment:
It should definitely be optional, long term we shouldn't need this at all. I need to get some clarification on what // +optional does exactly because I would assume this is optional as it stands right now, it can be created without.

Reply:
If it can be created without it, then it is optional. See https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#optional-vs-required.